InfoMovies/InfoMovies/Views/MainPage.xaml.cs

```diff
@@ -0,0 +1,111 @@
+﻿using InfoMovies.Helpers;
+using InfoMovies.ViewModels;
+using InfoMovies.Views;
+using System;
+using System.Collections;
+using Xamarin.Forms;
+
+namespace InfoMovies
+{
+    public partial class MainPage : BasePage
+    {
+        #region Fields
+
+        private MainPageViewModel viewModel;
+
+        #endregion
+
+        #region Constructors
+
+        public MainPage()
+        {
+            InitializeComponent();
+            BindingContext = viewModel = new MainPageViewModel();
+            InitializeItemsAsync();
+        }
+
+        #endregion
+
+        #region Overrides
+
+        protected override void OnAppearing()
+        {
+            base.OnAppearing();
+            LogHelper.TrackEvent("App initialized");
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private async void InitializeItemsAsync()=>
+            await viewModel.LoadItemsAsync();
+
+        private async void ListView_ItemAppearing(object sender, ItemVisibilityEventArgs e)
```

Comment:
Let's say that we need to use a infinity list in more than one page. How can you improve it so it can be reused?

Reply:
We could create a custom control, just like the horizontal list.
And to make it easier, it would create an asynchronous collection with the "MoveNext" and "FillList" commands to be implemented